File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
     
        int[] result=new int[2];
        
        int i=0;
        int j=nums.length-1;
        
       
        HashMap<Integer,Integer> temp=new HashMap<>();
        
        for(i=0;i<=j;i++){
           
            int diff=target-nums[i];
            
            if(temp.containsKey(diff)){
                result[0]=i;
                result[1]=temp.get(diff);
                return result;
            }
            
            temp.put(nums[i],i);
        }
        
        return result;
    }
}

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

/*
To read : 
1. Map - https://www.w3schools.com/js/js_maps.asp
2. Arrays - https://www.w3schools.com/js/js_arrays.asp

*/
var twoSum = function(nums, target) {
    let map=new Map();
    
    for(let i=0;i<nums.length;i++){
        
        if(map.has(target-nums[i])){
            return [map.get(target - nums[i]),i];
        }else {
            map.set(nums[i],i);
        }
        
    }
    
    return [];
};